examples/energy-management-app/energy-management-common/energy-reporting/include/ElectricalSensorInit.h

```diff
@@ -0,0 +1,33 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <ElectricalPowerMeasurementDelegate.h>
+#include <PowerTopologyDelegate.h>
+#include <lib/core/CHIPError.h>
+#include <lib/core/DataModelTypes.h>
+
+chip::app::Clusters::PowerTopology::PowerTopologyInstance * GetPTInstance();
+chip::app::Clusters::ElectricalPowerMeasurement::ElectricalPowerMeasurementInstance * GetEPMInstance();
```

Comment:
Since these are not widely used it would be good to leave inside the the place that inits everything and injects it down from the places that need it, rather than having to go through these global getters that init more globals in their guts.

Reply:
It may even be worthwhile to introduce a class to hold this state.

Reply:
Got rid of those. Now we have a a `CommonMain.h` which initializes the common clusters (DEM, PT, EPM, etc.) and then from there, it initializes either EVSE or WaterHeater and injects the references down, instead of using a global getter. Thanks for the suggestion!